pilot/pkg/networking/util/util.go

```diff
@@ -171,9 +171,20 @@ func AddrStrToCidrRange(addr string) (*core.CidrRange, error) {
 	}, nil
 }
 
+// OptionalBuildOptions is a struct to hold optional parameters for BuildAddress and BuildAdditionalAddresses.
+type OptionalBuildAddressOptions struct {
+	Transport istionetworking.TransportProtocol
+}
+
 // BuildAddress returns a SocketAddress with the given ip and port or uds.
-func BuildAddress(bind string, port uint32) *core.Address {
-	address := BuildNetworkAddress(bind, port, istionetworking.TransportProtocolTCP)
+func BuildAddress(bind string, port uint32, options ...OptionalBuildAddressOptions) *core.Address {
```

Comment:
I used a variadic variable to add extra build options for simplicity.
If you prefer I can add the transport socket_type as a normal parameter but I will have to update every existing use of BuildAddress and BuildAdditionalAddress to use TCP (except for that one time where QUIC listeners are needed).